e2e-tests/editor.spec.ts

```diff
@@ -247,282 +265,316 @@ test('copy and paste block after editing new block #5962', async ({ page, block
     await autocompleteMenu.expectVisible(modalName)
   }
 })
 
-test('press escape when autocomplete menu is open, should close autocomplete menu only #6270', async ({ page, block }) => {
-  for (const [commandTrigger, modalName] of [['[[', 'page-search'], ['/', 'commands']]) {
+test('press escape when autocomplete menu is open, should close autocomplete menu only #6270', async ({
+  page,
+  block,
+}) => {
+  for (const [commandTrigger, modalName] of [
+    ['[[', 'page-search'],
+    ['/', 'commands'],
+  ]) {
     await createRandomPage(page)
 
     // Open the action modal
-    await block.mustFill('text ')
-    await page.waitForTimeout(550)
-    await page.keyboard.type(commandTrigger, { delay: 20 })
-
-    await page.waitForTimeout(100)
+    await page.type(textAreaSelector, commandTrigger, { delay: 20 })
     await expect(page.locator(`[data-modal-name="${modalName}"]`)).toBeVisible()
-    await page.waitForTimeout(100)
 
     // Press escape; should close action modal instead of exiting edit mode
-    await page.keyboard.press('Escape')
-    await page.waitForTimeout(100)
-    await expect(page.locator(`[data-modal-name="${modalName}"]`)).not.toBeVisible()
-    await page.waitForTimeout(1000)
+    await page.waitForTimeout(30)
+    await page.keyboard.press('Escape', { delay: 10 })
+    await expect(
+      page.locator(`[data-modal-name="${modalName}"]`)
+    ).not.toBeVisible()
     expect(await block.isEditing()).toBe(true)
   }
 })
 
-test('press escape when link/image dialog is open, should restore focus to input', async ({ page, block }) => {
-  for (const [commandTrigger, modalName] of [['/link', 'commands']]) {
-    await createRandomPage(page)
-
-    // Open the action modal
-    await block.mustFill('')
-    await page.waitForTimeout(550)
-    await page.keyboard.type(commandTrigger, { delay: 20 })
-
-    await page.waitForTimeout(100)
-    await expect(page.locator(`[data-modal-name="${modalName}"]`)).toBeVisible()
-    await page.waitForTimeout(100)
-
-    // Press enter to open the link dialog
-    await page.keyboard.press('Enter')
-    await expect(page.locator(`[data-modal-name="input"]`)).toBeVisible()
+test('press escape when link/image dialog is open, should restore focus to input', async ({
+  page,
+  block,
+}) => {
+  let dataModalSelector: string // open modal/dialog selector
+  await createRandomPage(page)
 
+  for (const [commandTrigger, modalName] of [['/link', 'commands']]) {
+    // Step 1: Open & close the slash command menu
+    dataModalSelector = `[data-modal-name="${modalName}"]`
+    //! the space is needed to trigger the slash command menu in the CI environment
+    await page.type(textAreaSelector, ` ${commandTrigger}`, { delay: 20 })
+    // wait for the slash command menu to appear
+    await expect(page.locator(dataModalSelector)).toBeVisible()
+
+    // Step 2: Open & close the link dialog
+    dataModalSelector = `[data-modal-name="input"]`
```

Comment:
Also non blocking, but this could be a util, that can also be reused a couple of lines above and here https://github.com/logseq/logseq/pull/9594/files#diff-53fe4ba0c43b632e9781ee89d149a247b26ec25e88076a24eedf9b2f03ae96ecR19